packages/shopify-codemod/transforms/split-if-statement-assignment-parameter.js

```diff
@@ -0,0 +1,30 @@
+export default function splitIfStatementAssignmentParameter({source}, {jscodeshift: j}, {printOptions = {}}) {
+  const sourceAST = j(source);
+  sourceAST
+    .find(j.IfStatement, {
+      type: j.IfStatement.name,
+      test: {
+        type: j.AssignmentExpression.name,
+      },
+    }).forEach((nodePath) => {
+      const args = nodePath.get('test').node;
+      nodePath.insertBefore(j.expressionStatement(args));
+      nodePath.get('test').replace(args.left);
+
+    });
+  sourceAST
+    .find(j.IfStatement, {
+      test: {
+        type: j.SequenceExpression.name,
+        expressions: (expressions) => expressions.length > 1,
+      },
+    }).forEach((nodePath) => {
+      const args = nodePath.get('test').node;
+      args.expressions = args.expressions.map((assignment) => {
```

Comment:
I think it would be better to do something like this (if it's possible; you are modifying the expression in a loop so maybe not?)

```js
nodePath.get('test').forEach((expressionPath) => {
  nodePath.insertBefore(j.expressionStatement(assignment));
  expressionPath.replace(expressionPath.get('left').node);
});